### DISCUSSION TITLE
imagemagick → imagemagick-full

### DISCUSSION THREAD

**Discussion by tlk** [tlk/homebrew-imagemagick-x11#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas and engage with other community members.
  * Welcome others and are open-minded.

**Comment by pfitzner** [comment]
I installed version 7.1.2-15_1 and noticed a pile of delegates are missing, most notably `fontconfig`. Now I have no fonts unless I code in their full paths. Several other notable omissions too: jp2, jxl, openexr, lqr, gslib, ps, raw, and zstd are all gone. I'm hesitant to upgrade unless these problems are fixed. Otherwise, heartfelt thanks!

**Reply by tlk** [reply]
It looks like the homebrew core maintainers [decided to slim down the upstream imagemagick formula](https://github.com/Homebrew/homebrew-core/pull/263377) and introduced a new "full" formula: https://github.com/Homebrew/homebrew-core/commits/master/Formula/i/imagemagick-full.rb

Maybe we should track the new "full" formula moving forward?

**Reply by pfitzner** [reply]
Yes, indeed! It appears to restore the old functionality, and even to provide functionality beyond that of base ImageMagick, but I'm a fan of abundance. We really need to maintain support for X11, even though XQuartz misbehaves.

**Reply by tlk** [reply]
I have pushed an update to track the "full" formula. Can you check if this works as expected?

**Reply by pfitzner** [reply]
I'm afraid not. 

```
rwschaaf@narseh:~> brew install tlk/imagemagick-x11/imagemagick
==> Fetching downloads for: imagemagick
✔︎ Formula imagemagick (7.1.2-15)                                                                         Verified     10.8MB/ 10.8MB
==> Installing imagemagick from tlk/imagemagick-x11
==> Patching
==> ./configure --enable-osx-universal-binary=no --disable-silent-rules --disable-opencl --enable-shared --enable-static --with-gvc=
==> make install
==> Caveats
imagemagick-full includes additional tools and libraries that are not included in the regular imagemagick formula.
==> Summary
🍺  /opt/homebrew/Cellar/imagemagick/7.1.2-15_1: 809 files, 31.8MB, built in 2 minutes 7 seconds
==> Running `brew cleanup imagemagick`...
Disable this behaviour by setting `HOMEBREW_NO_INSTALL_CLEANUP=1`.
Hide these hints with `HOMEBREW_NO_ENV_HINTS=1` (see `man brew`).
Removing: /Users/rwschaaf/Library/Caches/Homebrew/imagemagick--7.1.2-15.tar.xz... (10.8MB)
rwschaaf@narseh:~> identify -version
Version: ImageMagick 7.1.2-15 Q16-HDRI aarch64 23686 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/license/
Features: Cipher DPC HDRI Modules
Delegates (built-in): bzlib freetype heic jng jpeg lcms ltdl lzma png tiff webp x xml zlib zstd
Compiler: clang (17.0.0)
```

It appears to be based on `imagemagick-full`, but the `fontconfig` delegate is still missing. Depending on full font pathnames guarantees non-portable code. If `fontconfig` isn't in `imagemagick-full`, perhaps the problem should be bumped up to the homebrew maintainers.

Of course, this depends on whether I'm making some bone-headed error!

**Reply by tlk** [reply]
I see, something does look odd. I would expect your homebrew to be installing [7.1.2-17](https://github.com/tlk/homebrew-imagemagick-x11/blob/faaf01856aebd18769e72e6bfb8f8e298c85106c/Formula/imagemagick.rb#L4) with fontconfig but it seems to be stuck at 7.1.2-15. You may have to run `brew upgrade`. Just be aware that it will upgrade all the software you have installed with homebrew.

```
tlk@5k ~ % identify -version
Version: ImageMagick 7.1.2-17 Q16-HDRI x86_64 23794 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/license/
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib cairo fontconfig freetype gslib heic jng jp2 jpeg jxl lcms lqr ltdl lzma openexr png ps raw rsvg tiff uhdr webp x xml zip zlib zstd
Compiler: clang (13.0.0)
tlk@5k ~ % 
```

**Reply by pfitzner** [reply]
All is well. It took `brew update && brew upgrade` to do the trick, and then `cargo clean`s for my Rust packages. "Bone-headed" is an apt characterization. Thanks a million!

**Reply by tlk** [reply]
You are welcome. Thanks for the confirmation!

**Comment by tlk** [comment]
(I have updated the title of this discussion to reflect its content)
